packages/eslint-plugin-pf-codemods/src/rules/v6/emptyStateHeaderMoveIntoEmptyState/emptyStateHeader-move-into-emptyState.ts

```diff
@@ -171,21 +78,56 @@ module.exports = {
         }
 
         const headingClassNameValue = getAttributeValueText(
+          context,
           headingClassNameAttribute
         );
 
         const headingClassName = headingClassNameValue
           ? `headerClassName=${headingClassNameValue}`
           : "";
-        const headingLevel = getAttributeText(headingLevelAttribute);
-        const titleClassName = getAttributeText(titleClassNameAttribute);
-        const titleTextPropValue = getAttributeText(titleTextAttribute);
+        const headingLevel = getAttributeText(context, headingLevelAttribute);
+        const titleClassName = getAttributeText(
+          context,
+          titleClassNameAttribute
+        );
+        const titleTextPropValue = getAttributeText(
+          context,
+          titleTextAttribute
+        );
+
+        const getChildrenText = (children: JSXElement["children"]) => {
+          if (!children.length) {
+            return "";
+          }
+
+          if (children.length === 1 && children[0].type === "JSXText") {
+            return `"${children[0].value.trim()}"`;
+          }
+
+          const potentialSingleChild = children.filter(
+            (child) => !(child.type === "JSXText" && child.value.trim() === "")
+          );
+
+          if (potentialSingleChild.length === 1) {
+            const singleChild = potentialSingleChild[0];
+            const singleChildText = context
+              .getSourceCode()
+              .getText(
+                singleChild as JSXExpressionContainer | JSXElement | JSXFragment
+              );
+
+            return singleChild.type === "JSXExpressionContainer"
+              ? singleChildText
+              : `{${singleChildText}}`;
+          }
+
+          return `{<>${getNodesText(context, children as Node[])}</>}`;
+        };
```

Comment:
Would it not make sense to extract this to your getText file?

Reply:
Probably yes, I at first thought it is too specific use case, but as I am looking at it, it can be extracted. I renamed it to `getChildrenAsAttributeValueText`. This name makes more sense to me, because it is transforming children to a text which is being used as an attribute value (including the expression container or quotation marks).